### DISCUSSION TITLE
[BUG] git clone always fails after first call in Windows

### DISCUSSION THREAD

**Discussion by dewyatt** [community/community#25078] [Actions]
<p><a href="https://github.com/dewyatt/gha-windows-git-clone" rel="nofollow noopener">Tiny example.</a></p>
<p><a href="https://github.com/dewyatt/gha-windows-git-clone/commit/f365a29b287a90b75fb1d662940a50bf42c6030b/checks?check_suite_id=392681757" rel="nofollow noopener">Log.</a></p>
<p>I emailed a bug report to support but didn’t hear back just yet and figured I ought to post here and see if anyone has experienced this.</p>
<p>Am I just lacking sleep?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Git command output are done on stderr not stdout, so you need to pipe stderr.   Set GIT_REDIRECT_STDERR=’2&gt;&amp;1’ for env first, code as below, please refer to the <a href="https://stackoverflow.com/questions/54372601/running-git-clone-from-powershell-giving-errors-even-when-it-seems-to-work" rel="nofollow noopener">link</a> for more details.</p>
<pre><code>steps:
      - uses: actions/checkout@v1
      - shell: powershell
        run: |
          set-strictmode -version 3.0
          set-psdebug -trace 2
          . .\tst.ps1
        env:
          GIT_REDIRECT_STDERR: '2&gt;&amp;1'
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5ed206c2abb76be823f0df91f1a70c5252b15baa.png" data-download-href="/uploads/short-url/dwORqtq1sdo20HYau7qujTeiBJ0.png?dl=1" title="GitStderr.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181088677-2133e6b5-ea85-44e1-88d5-c512adfc157e.png" alt="GitStderr.png" data-base62-sha1="dwORqtq1sdo20HYau7qujTeiBJ0" width="690" height="237" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5ed206c2abb76be823f0df91f1a70c5252b15baa_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GitStderr.png</span><span class="informations">700×241 14.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by dewyatt** [comment]
<p>Thank you much for the workaround! I see folks on Azure are having issues with this as well (<a href="https://github.com/microsoft/azure-pipelines-image-generation/issues/740" rel="nofollow noopener">here</a>).</p>
